Clarify how the mesh file formats are "processed" and improve doc.

Lucy-tri opened the pull request:
Remove example that uses DAE file which Drake doesn't process.
Remove example of SDF that used "model://" that Drake doesn't parse.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/robotlocomotion/drake/5507)
<!-- Reviewable:end -->

Lucy-tri commented:
+@liangfok for feature review.
+@afastman-tri  +@kunimatsu-tri (look for rgb) +@seancurtis-tri 
After this PR I will work on importing the YCB cup into a Drake OBJ file. And then update this some more (maybe concerning textures). 

---

Review status: 0 of 1 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-:-KfIvKGqhaQ2w1kQn9TZ:b-ly33t9)*
<!-- Sent from Reviewable.io -->

kunimatsu-tri commented:
*[drake/doc/models.rst, line 167 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfJN9XPo-z2NyUDLqPr:-KfJN9XQ0nqTCRlCNsBK:biuntjt) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L167)):*
> ```reStructuredText
> `RgbdCamera <http://drake.mit.edu/doxygen_cxx/rgbd__camera_8h.html>`_.
> Regardless of what program is processing the visual data, the processing time
> will be longer with a complex visual representation, than with a simple one, but
> ```

For the record, I compared rendering time for `ycb/013_apple/google_16k` and `ycb/013_apple/google_512k`.

It took 3.34s for `16k` while 8.59s for `512k`.
(Rendered a hundred times for both cases with the scene that has three objects with the same mesh file)

So, what you are writing is true.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5507)*
<!-- Sent from Reviewable.io -->

kunimatsu-tri commented:
*[drake/doc/models.rst, line 106 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfJPH4QUlv_nTo-s2nh:-KfJPH4QUlv_nTo-s2ni:b-lefzkh) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L106)):*
> ```reStructuredText
> If Drake processes an sdf or urdf file and finds a mesh file name extension that
> is not a type that it can process (currently only OBJ), it will see if a file
> exists with the same path and name but with an OBJ extension.
> ```

Is this a consensus for entire Drake?

If so, I need to change `RgbdCamera`'s implementation to comply with this.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5507)*
<!-- Sent from Reviewable.io -->

liangfok commented:
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[drake/doc/models.rst, line 106 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfJPH4QUlv_nTo-s2nh:-KfJYoI6m3fkIXvTe0tx:b-wcwznz) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L106)):*
<details><summary><i>Previously, kunimatsu-tri (Kunimatsu Hashimoto) wrote…</i></summary><blockquote>

Is this a consensus for entire Drake?

If so, I need to change `RgbdCamera`'s implementation to comply with this.
</blockquote></details>

Here is the relevant method:

https://github.com/RobotLocomotion/drake/blob/203a9a83dcd69a2a964230a4e6e071ac3f9e6224/drake/multibody/shapes/geometry.cc#L277

While this is the current behavior, I doubt it has a consensus. We should probably seek the dynamics team's roadmap on whether this will continue to persist after the transition to MultiBodyTree. \CC: @sherm1

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5507)*
<!-- Sent from Reviewable.io -->

sherm1 commented:
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[drake/doc/models.rst, line 106 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfJPH4QUlv_nTo-s2nh:-KfK-1g6rNxuAIQ00klq:br5rk8w) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L106)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

Here is the relevant method:

https://github.com/RobotLocomotion/drake/blob/203a9a83dcd69a2a964230a4e6e071ac3f9e6224/drake/multibody/shapes/geometry.cc#L277

While this is the current behavior, I doubt it has a consensus. We should probably seek the dynamics team's roadmap on whether this will continue to persist after the transition to MultiBodyTree. \CC: @sherm1
</blockquote></details>

I think this is just documenting the current behavior rather than a prescription for ideal behavior. Better would be to be able to read more formats! If RgbdCamera has the same obj-only limitation it might be wise to duplicate this behavior for consistency. We have no plans to change this behavior in conjunction with MBTree or GeometryWorld, but it wouldn't be difficult if it became urgent.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5507)*
<!-- Sent from Reviewable.io -->

Lucy-tri commented:
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[drake/doc/models.rst, line 106 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfJPH4QUlv_nTo-s2nh:-KfLsnXK20iisrobN3UH:bn2u4zb) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L106)):*
<details><summary><i>Previously, sherm1 (Michael Sherman) wrote…</i></summary><blockquote>

I think this is just documenting the current behavior rather than a prescription for ideal behavior. Better would be to be able to read more formats! If RgbdCamera has the same obj-only limitation it might be wise to duplicate this behavior for consistency. We have no plans to change this behavior in conjunction with MBTree or GeometryWorld, but it wouldn't be difficult if it became urgent.
</blockquote></details>

The behavior in geometry.cc is so strange and obscure, that Liang and I felt it should be documented here, also because there are several DAE, STL and WRL files checked into Drake, which would leave one to believe that Drake can "process" those file types.

---

*[drake/doc/models.rst, line 167 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfJN9XPo-z2NyUDLqPr:-KfLtrSngoAxS-Kf1gkB:b-ucfknh) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L167)):*
<details><summary><i>Previously, kunimatsu-tri (Kunimatsu Hashimoto) wrote…</i></summary><blockquote>

For the record, I compared rendering time for `ycb/013_apple/google_16k` and `ycb/013_apple/google_512k`.

It took 3.34s for `16k` while 8.59s for `512k`.
(Rendered a hundred times for both cases with the scene that has three objects with the same mesh file)

So, what you are writing is true.


</blockquote></details>

Thanks!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5507)*
<!-- Sent from Reviewable.io -->

kunimatsu-tri commented:
*[drake/doc/models.rst, line 106 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfJPH4QUlv_nTo-s2nh:-KfMEIcZZ5lk4qp6KlFs:bddwc9d) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L106)):*
<details><summary><i>Previously, Lucy-tri (Lucy) wrote…</i></summary><blockquote>

The behavior in geometry.cc is so strange and obscure, that Liang and I felt it should be documented here, also because there are several DAE, STL and WRL files checked into Drake, which would leave one to believe that Drake can "process" those file types.
</blockquote></details>

Thanks all.  Good to know and documenting the non-straight forward behavior makes sense even if it's not the ideal one.  Then, I will change `RgbdCamera`'s behavior to be the same as this.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5507)*
<!-- Sent from Reviewable.io -->

kunimatsu-tri commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-:-KfMFvO0vqduokgRPWPy:bnfp4nl)*
<!-- Sent from Reviewable.io -->

liangfok commented:
Completed first pass.

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 15 unresolved discussions.

---

*[drake/doc/models.rst, line 104 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfMG0Ghv_EcQgZ1j1He:-KfJ_7XpQtJscnT_HXA7:bjhir7e) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L104)):*
> ```reStructuredText
>   definition file format.
> 
> If Drake processes an sdf or urdf file and finds a mesh file name extension that
> ```

BTW: nit: "sdf" and "urdf" should be "SDF" and "URDF", respectively, since they are acronyms and to be consistent with the rest of this document.

---

*[drake/doc/models.rst, line 106 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfMGCZ7zyx7Z3jOVdCR:-KfMGCZ7zyx7Z3jOVdCS:b-t49ycg) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L106)):*
Consider replacing:

> with an OBJ extension

with:

> with a type that it can process.

to be more robust against the addition of support for future mesh types.

---

*[drake/doc/models.rst, line 109 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfMHAbbk52G_ZC-Y-We:-KfMHAbbk52G_ZC-Y-Wf:b3lt2g2) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L109)):*
> ```reStructuredText
> 
> The following are mesh file formats that are not processed by Drake, but may be
> found in the Drake repository and may be used to find OBJ versions:
> ```

ditto regarding changing "OBJ versions" to "compatible versions" or "processable versions".

---

*[drake/doc/models.rst, line 123 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfMHhcbTjP-Z-L3GMGd:-KfMHhcbTjP-Z-L3GMGe:brah176) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L123)):*
BTW, consider deleting ", in the form of a text file" since that seems to be generally true of all mesh formats.

Consider replacing it with:

> that is superceded by X3D but is still in use today.

---

*[drake/doc/models.rst, line 145 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfMINORnkomnvW6orQN:-KfMINORnkomnvW6orQO:brzzb4e) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L145)):*
> ```reStructuredText
> --------------------------------
> 
> To model a shape for visual purposes, use the ``<visual>`` tag. To model a shape
> ```

BTW, consider replacing "the" with "URDF's or SDF's".

---

*[drake/doc/models.rst, line 147 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfMIVgY-eH0yV3VKglB:-KfMIVgY-eH0yV3VKglC:brzzb4e) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L147)):*
> ```reStructuredText
> To model a shape for visual purposes, use the ``<visual>`` tag. To model a shape
> for the purpose of determining contact between your model and other objects, use
> the ``<collision>`` tag.
> ```

BTW, consider replacing "the" with "URDF's or SDF's".

---

*[drake/doc/models.rst, line 153 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfMIerHyHU_5xkYwaCg:-KfMIerHyHU_5xkYwaCh:b-k6xdvp) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L153)):*
> ```reStructuredText
> 
> An object's shape can be modelled using a 3D scanner, which produces a polygon
> mesh. Meshes of complicated objects contain many polygons. The detail (density)
> ```

BTW, consider adding "geometrically" before "complicated"

---

*[drake/doc/models.rst, line 161 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfMIsWXmDMOZzlFMkGH:-KfMIsWXmDMOZzlFMkGI:b-5kvss) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L161)):*
> ```reStructuredText
> 
> The ``<visual>`` tag is used in visualization programs like
> drake-visualizer (in the
> ```

BTW, consider surrounding `drake-visualizer` in double-back-ticks to emphasize that it's an executable.

---

*[drake/doc/models.rst, line 168 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfMJBYXiEdz4NcaV_nv:-KfMJBYXiEdz4NcaV_nw:bdzxpsp) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L168)):*
> but not as much as if it were collision data.

This statement may not be true if the collision model were for some reason more complicated than the visual model.

---

*[drake/doc/models.rst, line 170 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfMJTJViRquHGdkVtaM:-KfMJOSHug2osLWxWkC5:b-m8vlzb) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L170)):*
> ```reStructuredText
> not as much as if it were collision data.
> 
> In addition to modelling shapes with meshes, you can also model a shape with
> ```

BTW, consider replacing "a shape" with "shapes" to be consistent with the plurality of the first part of the sentence.

---

*[drake/doc/models.rst, line 172 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfMJgWe42Ff-X33dIW4:-KfMJgWe42Ff-X33dIW5:bkc4m0e) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L172)):*
> ```reStructuredText
> In addition to modelling shapes with meshes, you can also model a shape with
> geometric primitives, such as cylinders, spheres, or boxes. The complexity of a
> geometric primitive is far less than with a mesh.
> ```

BTW, consider adding ", which results in lower processing time" to the end of this sentence.

---

*[drake/doc/models.rst, line 179 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfMJxWcurDSw0dRP1Ok:-KfMJxWcurDSw0dRP1Ol:bqkd87i) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L179)):*
> The idea is that for links of the arm, we don't need exact shapes
> for collision, where we don't expect to need precision.

Consider rewording:

> This is because we don't anticipate the links of the arm to collide with other
> objects, and thus don't need the extra modeling precision offered by meshes.

---

*[drake/doc/models.rst, line 185 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfMKSkhybgTyWEDbaF1:-KfMKSkhybgTyWEDbaF2:b-kfzy9y) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L185)):*
> But there are cases where we need the precision of a mesh, for example, for the
> end of the arm, when the arm is reaching into tight or crowded spaces or
> performing dexterous manipulation tasks.

Consider splitting and clarifying this sentence:

> But there are cases where we need the precision of a mesh. For example, extra modeling precision may be needed for a robot arm's end effector to simulate a scenario where it is reaching into tight spaces or performing dexterous manipulation tasks.

---

*[drake/doc/models.rst, line 186 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfMKvEYP2fImaBPmeaC:-KfMKvEYP2fImaBPmeaD:b-aiv2lu) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L186)):*
> In these cases, we need to simplify the mesh.

Consider clarifying:

> In these cases, the processing overhead may become too large, resulting in the need to simplify the mesh.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-:-KfMLC5ZskfePBLWDNxT:bl4rucz)*
<!-- Sent from Reviewable.io -->

SeanCurtis-TRI commented:
Review status: all files reviewed at latest revision, 17 unresolved discussions.

---

*[drake/doc/models.rst, line 106 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfMMiAJum53NrVqOXrW:-KfMMiAJum53NrVqOXrX:ba083bw) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L106)):*
> ```reStructuredText
> If Drake processes an sdf or urdf file and finds a mesh file name extension that
> is not a type that it can process (currently only OBJ), it will see if a file
> exists with the same path and name but with an OBJ extension.
> ```

nit: Given the case sensitivity of linux and the fact the code *explicitly* replaces the extension with a *lowercase* `obj` extension, we might want to be more careful about how we talk about "OBJ extension".

---

*[drake/doc/models.rst, line 109 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfMHAbbk52G_ZC-Y-We:-KfMO46j8prdp39XPAHx:b-50e4ar) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L109)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

ditto regarding changing "OBJ versions" to "compatible versions" or "processable versions".
</blockquote></details>

BTW My thought on this is that @liangfok's suggestion has nice generality.  But since the current state of the code is so specific, being explicit about that limited behavior makes sense.  If/when we add more valid formats, the general language would make more sense.

---

*[drake/doc/models.rst, line 150 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfMOaC-Q8bPrx197duw:-KfMOaC-Q8bPrx197dux:b5qgc0c) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L150)):*
> ```reStructuredText
> 
> To explain why you might want to define the collision element differently than
> the visual element, let's explore the different ways of defining shapes.
> ```

BTW It might be worth noting that while URDF and SDF both use `<collision>` and `<visual>` tags, that the structure and contents of those tags are not necessarily the same.  You can refer them to the docs for the details, but the language as it currently stands may imply more of a parallel than actually exists.

---

*[drake/doc/models.rst, line 168 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfMJBYXiEdz4NcaV_nv:-KfMPEAIoj_75Okl_b00:b-hle542) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L168)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

> but not as much as if it were collision data.

This statement may not be true if the collision model were for some reason more complicated than the visual model.
</blockquote></details>

BTW I get the sense that the message here is, for a given level of complexity, the visual operations are still cheaper than the collision operations.  If that's the case, it could be clarified.  Although, I would advocate just removing the final contrast; I don't know how much value it provides.  The point has already been made that simple geometry is better for collision.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5507)*
<!-- Sent from Reviewable.io -->

liangfok commented:
Review status: all files reviewed at latest revision, 16 unresolved discussions.

---

*[drake/doc/models.rst, line 109 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfMHAbbk52G_ZC-Y-We:-KfMQEVJw7A5-dZKDtic:b-79b3gc) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L109)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

BTW My thought on this is that @liangfok's suggestion has nice generality.  But since the current state of the code is so specific, being explicit about that limited behavior makes sense.  If/when we add more valid formats, the general language would make more sense.
</blockquote></details>

Agreed. I'll mark myself satisfied.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5507)*
<!-- Sent from Reviewable.io -->

SeanCurtis-TRI commented:
<img class="emoji" title=":LGTM:" alt=":LGTM:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> with a nit

---

Review status: 0 of 1 files reviewed at latest revision, 15 unresolved discussions.

---

*[drake/doc/models.rst, line 148 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfNCKRJet8WJPB1n56A:-KfNCKRJet8WJPB1n56B:bbe4cv) ([raw file](https://github.com/robotlocomotion/drake/blob/21e292d17785b703df685faa4752ce466e0a83b8/drake/doc/models.rst#L148)):*
> ```reStructuredText
> model a shape for the purpose of determining contact between your model and
> other objects, use URDF's or SDF's ``<collision>`` tag. (While both SDF and URDF
> use those same tags, note that the structure and content of those tags differ).
> ```

nit: When the whole sentence is parenthesized, the period goes on the inside.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-:-KfNC_--p8Z0HGjPyP7V:bo3felr)*
<!-- Sent from Reviewable.io -->

SeanCurtis-TRI commented:
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 15 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5507)*
<!-- Sent from Reviewable.io -->

Lucy-tri commented:
Review status: all files reviewed at latest revision, 17 unresolved discussions.

---

*[drake/doc/models.rst, line 104 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfMG0Ghv_EcQgZ1j1He:-KfNBzJXEYGEeUgUn7uh:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L104)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW: nit: "sdf" and "urdf" should be "SDF" and "URDF", respectively, since they are acronyms and to be consistent with the rest of this document.
</blockquote></details>

Done.

---

*[drake/doc/models.rst, line 106 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfMGCZ7zyx7Z3jOVdCR:-KfMkJvExUG59t9YToK-:b-cw3lja) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L106)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

Consider replacing:

> with an OBJ extension

with:

> with a type that it can process.

to be more robust against the addition of support for future mesh types.
</blockquote></details>

I think your proposed change is less clear and more confusing, because it implies there are other types that it can process besides OBJ.  We'll have to review this again when we add true support for other mesh types. 

---

*[drake/doc/models.rst, line 106 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfMMiAJum53NrVqOXrW:-KfNCI8ZYR38umrVR-JV:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L106)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

nit: Given the case sensitivity of linux and the fact the code *explicitly* replaces the extension with a *lowercase* `obj` extension, we might want to be more careful about how we talk about "OBJ extension".
</blockquote></details>

Done.

---

*[drake/doc/models.rst, line 123 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfMHhcbTjP-Z-L3GMGd:-KfNCMhmrLfJ0PrmObnd:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L123)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW, consider deleting ", in the form of a text file" since that seems to be generally true of all mesh formats.

Consider replacing it with:

> that is superceded by X3D but is still in use today.
</blockquote></details>

Done.

---

*[drake/doc/models.rst, line 145 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfMINORnkomnvW6orQN:-KfNCOV3SkqWBiiq2UoY:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L145)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW, consider replacing "the" with "URDF's or SDF's".
</blockquote></details>

Done.

---

*[drake/doc/models.rst, line 147 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfMIVgY-eH0yV3VKglB:-KfNCQ_oGk2twD1pSnGy:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L147)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW, consider replacing "the" with "URDF's or SDF's".
</blockquote></details>

Done.

---

*[drake/doc/models.rst, line 150 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfMOaC-Q8bPrx197duw:-KfNCS-fP-exnUtyR9qo:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L150)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

BTW It might be worth noting that while URDF and SDF both use `<collision>` and `<visual>` tags, that the structure and contents of those tags are not necessarily the same.  You can refer them to the docs for the details, but the language as it currently stands may imply more of a parallel than actually exists.
</blockquote></details>

Done.

---

*[drake/doc/models.rst, line 153 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfMIerHyHU_5xkYwaCg:-KfNCSz3yhfdnFBrAhMA:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L153)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW, consider adding "geometrically" before "complicated"
</blockquote></details>

Done.

---

*[drake/doc/models.rst, line 161 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfMIsWXmDMOZzlFMkGH:-KfNCUQ-GQnNoxFkJ5fh:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L161)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW, consider surrounding `drake-visualizer` in double-back-ticks to emphasize that it's an executable.
</blockquote></details>

Done.

---

*[drake/doc/models.rst, line 168 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfMJBYXiEdz4NcaV_nv:-KfNCb05gBlJBgtCbz_w:bxk3q0x) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L168)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

BTW I get the sense that the message here is, for a given level of complexity, the visual operations are still cheaper than the collision operations.  If that's the case, it could be clarified.  Although, I would advocate just removing the final contrast; I don't know how much value it provides.  The point has already been made that simple geometry is better for collision.
</blockquote></details>

Rewrote.

---

*[drake/doc/models.rst, line 170 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfMJTJViRquHGdkVtaM:-KfNCeQGQb29LWp0f43R:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L170)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW, consider replacing "a shape" with "shapes" to be consistent with the plurality of the first part of the sentence.
</blockquote></details>

Done.

---

*[drake/doc/models.rst, line 172 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfMJgWe42Ff-X33dIW4:-KfNCgzAsdvkw3qRPEdX:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L172)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW, consider adding ", which results in lower processing time" to the end of this sentence.
</blockquote></details>

Done.

---

*[drake/doc/models.rst, line 179 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfMJxWcurDSw0dRP1Ok:-KfN6y9V4XRuIgLfaVeY:bz2y2sw) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L179)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

> The idea is that for links of the arm, we don't need exact shapes
> for collision, where we don't expect to need precision.

Consider rewording:

> This is because we don't anticipate the links of the arm to collide with other
> objects, and thus don't need the extra modeling precision offered by meshes.
</blockquote></details>

Oh, but we absolutely do care if the links collide with other objects, because we need to move the arm through spaces with walls and columns and such, and also in the example I gave with the elbow. Its the precision we don't need, in a link/arm. 

---

*[drake/doc/models.rst, line 185 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfMKSkhybgTyWEDbaF1:-KfNCn4YbIvJUCwv9blU:bhjexzp) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L185)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

> But there are cases where we need the precision of a mesh, for example, for the
> end of the arm, when the arm is reaching into tight or crowded spaces or
> performing dexterous manipulation tasks.

Consider splitting and clarifying this sentence:

> But there are cases where we need the precision of a mesh. For example, extra modeling precision may be needed for a robot arm's end effector to simulate a scenario where it is reaching into tight spaces or performing dexterous manipulation tasks.
</blockquote></details>

Rewrote and used most of what you suggested.

---

*[drake/doc/models.rst, line 186 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfMKvEYP2fImaBPmeaC:-KfNCypFHlja8mAmXhem:b-grx96e) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L186)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

> In these cases, we need to simplify the mesh.

Consider clarifying:

> In these cases, the processing overhead may become too large, resulting in the need to simplify the mesh.

</blockquote></details>

Rewrote with inspiration from Liang.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5507)*
<!-- Sent from Reviewable.io -->

liangfok commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[drake/doc/models.rst, line 185 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfMKSkhybgTyWEDbaF1:-KfNDozz4CkLAnV_n1td:bp0mu6w) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L185)):*
<details><summary><i>Previously, Lucy-tri (Lucy) wrote…</i></summary><blockquote>

Rewrote and used most of what you suggested.
</blockquote></details>

BTW: nit: "its" should be "it's".

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-:-KfNDxze5sc2Ns73Zim-:bnfp4nl)*
<!-- Sent from Reviewable.io -->

liangfok commented:
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5507)*
<!-- Sent from Reviewable.io -->

Lucy-tri commented:
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[drake/doc/models.rst, line 185 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfMKSkhybgTyWEDbaF1:-KfNGbUKVJWMeJat0-gS:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/406eb057db4708121fb14228396e12426eceeab6/drake/doc/models.rst#L185)):*
<details><summary><i>Previously, liangfok (Chien-Liang Fok) wrote…</i></summary><blockquote>

BTW: nit: "its" should be "it's".
</blockquote></details>

Done.

---

*[drake/doc/models.rst, line 148 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-KfNCKRJet8WJPB1n56A:-KfNGZp-Ii20kban_8jT:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/21e292d17785b703df685faa4752ce466e0a83b8/drake/doc/models.rst#L148)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

nit: When the whole sentence is parenthesized, the period goes on the inside.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5507)*
<!-- Sent from Reviewable.io -->

liangfok commented:
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5507)*
<!-- Sent from Reviewable.io -->

Lucy-tri commented:
+@jwnimmer-tri  for platform review, please.

---

Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-:-KfNHAqtd-loxxYFIBAt:b59ib8v)*
<!-- Sent from Reviewable.io -->

jwnimmer-tri commented:
Sphinx-only change with lots of feature review already.  <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> without reading it.

---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-:-KfNLPA7UHUU6cBu8ceH:bd7uy1a)*
<!-- Sent from Reviewable.io -->

Lucy-tri commented:
-@afastman-tri  

---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-:-KfNM0-AW5vVQDDMeW8m:bjssfbn)*
<!-- Sent from Reviewable.io -->

Lucy-tri commented:
+@afastman-tri 

---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-:-KfNNT2Ud4qCSiXIVo8g:b-jfs85x)*
<!-- Sent from Reviewable.io -->

afastman-TRI commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5507#-:-KfNVTOuDuiZ2Hl2YnkJ:bnfp4nl)*
<!-- Sent from Reviewable.io -->
